File: src/CommunityCar.Domain/Entities/Car.cs
namespace CommunityCar.Domain.Entities;

public class Car : BaseEntity
{
    public string Make { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public int Year { get; set; }
    public string LicensePlate { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal PricePerDay { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsAvailable { get; set; } = true;
    public string Location { get; set; } = string.Empty;
    
    public Guid OwnerId { get; set; }
    public User Owner { get; set; } = null!;
    
    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}

File: Mobile/lib/models/car.dart
class Car {
  final String id;
  final String make;
  final String model;
  final int year;
  final String licensePlate;
  final String? description;
  final double pricePerDay;
  final String? imageUrl;
  final bool isAvailable;
  final String location;
  final String ownerId;

  Car({
    required this.id,
    required this.make,
    required this.model,
    required this.year,
    required this.licensePlate,
    this.description,
    required this.pricePerDay,
    this.imageUrl,
    required this.isAvailable,
    required this.location,
    required this.ownerId,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'],
      make: json['make'],
      model: json['model'],
      year: json['year'],
      licensePlate: json['licensePlate'],
      description: json['description'],
      pricePerDay: (json['pricePerDay'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      isAvailable: json['isAvailable'],
      location: json['location'],
      ownerId: json['ownerId'],
    );
  }
}
